dev/com.ibm.ws.security.javaeesec.cdi/src/com/ibm/ws/security/javaeesec/cdi/beans/CustomFormAuthenticationMechanism.java

```diff
@@ -242,11 +263,23 @@ protected void createLoginHashMap(Subject clientSubject, CredentialValidationRes
 
     private IdentityStoreHandler getIdentityStoreHandler() {
         IdentityStoreHandler identityStoreHandler = null;
-        Instance<IdentityStoreHandler> storeHandlerInstance = CDI.current().select(IdentityStoreHandler.class);
+        Instance<IdentityStoreHandler> storeHandlerInstance = getCDI().select(IdentityStoreHandler.class);
         if (storeHandlerInstance.isUnsatisfied() == false && storeHandlerInstance.isAmbiguous() == false) {
             identityStoreHandler = storeHandlerInstance.get();
         }
         return identityStoreHandler;
     }
 
+    private boolean isSupportedCredential(@Sensitive Credential cred) {
```

Comment:
What happens if their identity store support other credential types and they want to use those?

Reply:
The current implementation of CustomForm only accepts credentials which takes user id and password. Therefore it has this check. 
However, as you stated if the application bundles it's own custom login bean which constructs some unique credential and identitystore which also can accept this unique credential, then this logic needs to be removed. The code was modified.